stock-tracker/src/socket/eventActions.ts

```diff
@@ -0,0 +1,86 @@
+import { ActionCreator } from "redux";
+import * as EventType from "./eventTypes";
+import { setChartDataAction, setChartErrorAction } from "features/chart";
+import {
+  setCompanyOverviewAction,
+  setErrorOverviewAction
+} from "features/overview";
+import { setLatestNewsAction, setErrorNewsAction } from "features/latestNews";
+import { setResponseAction, setErrorKeyStatsAction } from "features/keyStats";
+import { addTopPeersAction, setErrorPeersAction } from "features/topPeers";
+import { setSuggestionsAction, setErrorSearchAction } from "features/search";
+
+import { SetChartData, SetChartError } from "features/chart/redux/actions";
+import {
+  SetKeyStatsResponse,
+  SetKeyStatsError
+} from "features/keyStats/redux/actions";
+import {
+  SetLatestNews,
+  SetLatestNewsError
+} from "features/latestNews/redux/actions";
+import {
+  SetCompanyOverview,
+  SetCompanyOverviewError
+} from "features/overview/redux/actions";
+import { SetSuggestions, SetSearchError } from "features/search/redux/actions";
+import { AddTopPeers, SetTopPeersError } from "features/topPeers/redux/actions";
+
+export interface EventActionInterface {
+  event:
+    | typeof EventType.CHART_DATA
```

Comment:
q: I'm a bit unsure about the type safety of this, as you're creating a union of `action` and `errorAction` to all the possible actions, so you could technically mix `SetChartData` with `SetKeyStatsError`